### PR TITLE
Create EventFlow.Sql for generic SQL services

### PR DESCRIPTION
**Pull request by rasmus**
@JC008 I created this branch to implement the rename of `EventFlow,MsSql` and made changes that enables it to be used by e.g. SQLite in #150.

### CODE REVIEW

**Comment by JC008** [comment]
@rasmus I have started on `EventFlow.SQLite' for #150 and need to check if the error codes in `SqlErrorRetryStrategy` are universal across ADO providers or if SQLite specific code must be used.

**Comment by rasmus** [comment]
@JC008 I made it possible for SQLite to have its own retry strategy and configuration in the PR as I didn't think the error codes were common

**Comment by rasmus** [comment]
@JC008 I you think this solution would work, then I'll get it done and get it to `develop`. Then, unless you are doing changes locally, I'll merge develop into the SQLite branch.

File: Source/EventFlow.Sql/Integrations/TableParameter.cs [hunk 1]
```diff
@@ -31,7 +32,7 @@
 using Dapper;
 using Microsoft.SqlServer.Server;
```
Comment (JC008) [inline Source/EventFlow.Sql/Integrations/TableParameter.cs]: Dependency on `Microsoft.SqlServer.Server` should not be allowed.

Comment (rasmus) [inline Source/EventFlow.Sql/Integrations/TableParameter.cs]: @JC008 Very, very good point... Then I'll have to reintroduce the EventFlow.MsSql package once again and move the table parameter to that.

**Comment by JC008** [comment]
@rasmus Maybe its worth while adding another level of abstraction now to enable implementations (for `ReadStores` and `EventStores`) with NoSQL and Cloud storage systems later (e.g. Azure table store)?

**Comment by rasmus** [comment]
@JC008 I don't follow that abstraction would be? There's already implementations for Elasticsearch as a read store and EventStore as another event store and those don't rely on anything but the core EventFlow package and the specific client libraries.

**Comment by JC008** [comment]
@rasmus OK - ignore that thought then - I have not worked with those implementations yet.

**Pull request merged**